project/project-model.js

```diff
@@ -0,0 +1,39 @@
+const knexConfig = require('../knexfile');
+const knex = require('knex');
+
+const db = knex(knexConfig.development);
+
+module.exports = {
+  addProject,
+  getProject,
+};
+
+function addProject(project) {
+  return db('project')
+    .insert(project);
+}
+
+function getProject(id) {
+  return db('project')
+    .join('action', 'project.id', 'action.project_id')
+    .select('project.id', 'project.name', 'project.description', 'project.completed', 'action.id as action_id', 'action.description as action_description',
+      'action.notes as action_notes', 'action.completed as action_completed')
+    .where('project.id', id)
+    .reduce((accumulator, currentObj) => {
+
+      accumulator.id = currentObj.id;
+      accumulator.name = currentObj.name,
+      accumulator.description = currentObj.description,
+      accumulator.completed = currentObj.completed,
+
+      accumulator.actions.push({id:currentObj.action_id,
+        description:currentObj.action_description,
+        notes:currentObj.action_notes,
+        completed:currentObj.action_completed
+      });
+
+      return accumulator;
+    }, { actions: [] });
```

Comment:
We took care of refactoring this during our 1:1, just leaving this here for reference
```
router.get("/:id", async (req, res) => {
  const {id} = req.params;

  const query1 = await db("projects").where({id});
  const query2 = await db("actions as a").where("a.project_id", id);

  const response = {
    ...query1[0],
    actions: query2
  };

  db("projects")
    .where({id})
    .then(projects => res.status(200).json(response))
    .catch(err => res.status(500).json(err));
});
```

Still, impressive use of `.reduce()`